### PR TITLE
Don't panic when RunToPR gets a non-numeric PR number

### PR DESCRIPTION
**Pull request by Katharine**
This fixes the issue that led to spyglass not having symlink resolution leading to a panic instead of an error.

The panic is gone now because #11614 implemented symlink resolution, but this PR fixes the panic. Amusingly, the error was already handled, but creating the error caused a panic.

Also fix a similar bug I noticed while working on this.

### CODE REVIEW

File: prow/spyglass/spyglass.go [hunk 1]
```diff
@@ -248,9 +248,12 @@ func (s *Spyglass) RunToPR(src string) (string, string, int, error) {
 		if logType == gcs.NonPRLogs {
 			return "", "", 0, fmt.Errorf("not a PR URL: %q", key)
 		} else if logType == gcs.PRLogs {
+			if len(split) < 3 {
+				return "", "", 0, fmt.Errorf("malformed %s key %q should have at least three components", gcs.PRLogs, key)
+			}
 			prNum, err := strconv.Atoi(split[len(split)-3])
 			if err != nil {
-				return "", "", 0, fmt.Errorf("couldn't parse PR number %q in %q: %v", split[5], key, err)
+				return "", "", 0, fmt.Errorf("couldn't parse PR number %q in %q: %v", split[len(split)-3], key, err)
```
Comment (stevekuznetsov) [inline prow/spyglass/spyglass.go]: Can we use a variable for `split[len(split)-3]` to DRY out that expression in this stanza?

**Comment by stevekuznetsov** [comment]
/lgtm
/hold
one tiny nit

**Comment by stevekuznetsov** [comment]
Up to you if you care for it

File: prow/spyglass/spyglass.go (hunk 1 above)
Comment (Katharine) [reply prow/spyglass/spyglass.go]: Probably a good idea, given getting it wrong is the origin of this bug.

**Pull request merged**